spray-can/src/main/scala/spray/can/Http.scala

```diff
@@ -55,11 +57,13 @@ object Http extends ExtensionKey[HttpExt] {
                   endpoint: InetSocketAddress,
                   backlog: Int,
                   options: immutable.Traversable[Inet.SocketOption],
-                  settings: Option[ServerSettings])(implicit val sslEngineProvider: ServerSSLEngineProvider) extends Command
+                  settings: Option[ServerSettings],
+                  setupPipeline: Pipeline ⇒ Pipeline)(implicit val sslEngineProvider: ServerSSLEngineProvider) extends Command
```

Comment:
This is an usability extra, no needed at all to provide the functionality: What about adding a method to build a new `Bind` instance with composition over `setupPipeline` ? 

```scala
def updatePipelineWith(step: Pipeline ⇒ Pipeline): Bind = {
  copy(setupPipeline andThen step)
}
```

Users of this contract could do things like:

```scala
val bind: Bind = ...
val newBind = bind updatePipelineWith { pipeline =>
  ???
}
```